src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const sep = String(separator);
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === (this.length - 1) && !this[i]) {
+        return result;
+      } else if (this[i] === null || this[i] === undefined) {
+        result += sep;
+      } else if ((i + 1) !== this.length) {
+        result += this[i] + sep;
+      } else {
+        result += this[i];
+      }
```

Comment:
try to avoid `if else` constructions where can be used just flat `if`s

Reply:
simplify this query chain. you only need two checks: 
- null & undefined
- should we add the separator